draft-ietf-ccwg-rfc5033bis.md

```diff
@@ -282,7 +282,32 @@ flows using that algorithm share a bottleneck link, with no other algorithms
 operating.
 
 ### Protection Against Congestion Collapse
-
+The alternate congestion control mechanism should not cause increased
+overhead under adverse network conditions.
+
+This criteria can be evaluated by counting the total bytes (data and headers)
+arriving at the receiver when delivering a
+fixed workload over varying network conditions.
+The total delivered bytes should remain constant,
+independent of network conditions over the entire
+operating range for the protocol.
+For general transport protocols such as TCP
+this means over any path in the Internet.
+
+There are known exceptions to this guideline,
+for example spurious Retransmission
+Timeouts {{?RFC8298}}{{?RFC7765}} and TCP loss Probe {{?RFC8985}}.
+These algorithms can substantially improve application performance
+in certain environments at the cost of additional network overhead
+due to spurious retransmissions.
+
+Alternate congestion control mechanisms should be carefully
+evaluated for exceptions.
+That evaluation should appropriately reflect on any the inherent trade-offs.
+
+This criteria can also be applied protocol layers above transport.
+
```

Comment:
I remember several cases of congestion collapse in the late 80's -- we were interconnecting Ethernet networks by using X.25 virtual circuits, and things were not going well. But I would not describe the cause of the collapse as "increased overhead" -- unless we get a rather all-encompassing definition of overhead.

The immediate cause of the collapse was that excess traffic was causing queues, queues were causing the pre-1988 timers to fire, and TCP was immediately resending all the "lost" packets using a Go-back N strategy. In theory, that should only have affected the queue at a bottleneck, but in practice the excess traffic caused queues in nodes between senders and bottleneck, thus propagating the gridlock. It was nasty.

What got us out of that mess was of course Van Jacobson's congestion control algorithm, and also the work on improved TCP timers. But the principle being that was "if the conditions are bad, nodes should reduce their traffic, not increase it."

Reply:
So that might be a bug in my text: it should be all encompassing overhead, although the specifics of the text make it pretty clear.

Reply:
Ar least in the X.25 networks I used, the excess traffic also caused queuing packets in nodes under flow-control/loss recovery. This made MUCH worse by delaying packets carrying routing and other network functions. You might have said this was due to the layering (transport carrying network control traffic as equals) - importnatly that part was mitigated by diffserv to prioristise network critical traffic. The buildup of traffic  was best mitigated by backoff from deadlock - increasing timers and no regenerative retransmission.